XenAdmin/Diagnostics/Checks/PVGuestsCheck.cs

```diff
@@ -45,39 +46,41 @@ class PVGuestsCheck : HostPostLivenessCheck
         private readonly Pool _pool;
         private readonly bool _upgrade;
         private readonly Dictionary<string, string> _installMethodConfig;
-        private readonly bool _manualUpgrade;
-        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
 
-        public PVGuestsCheck(Pool pool, bool upgrade, Dictionary<string, string> installMethodConfig = null, bool manualUpgrade = false)
+        public PVGuestsCheck(Pool pool, bool upgrade, Dictionary<string, string> installMethodConfig = null)
             : base(Helpers.GetMaster(pool.Connection))
         {
             _pool = pool;
             _upgrade = upgrade;
             _installMethodConfig = installMethodConfig;
-            _manualUpgrade = manualUpgrade;
         }
 
         protected override Problem RunHostCheck()
         {
-            string upgradePlatformVersion;
             if (!_pool.Connection.Cache.VMs.Any(vm => vm.IsPvVm()))
                 return null;
-            if (!_upgrade || _manualUpgrade)
+
+            if (!_upgrade)
                 return new PoolHasPVGuestWarningUrl(this, _pool);
-            try
-            {
-                var result = Host.call_plugin(Host.Connection.Session, Host.opaque_ref, "prepare_host_upgrade.py", "getVersion", _installMethodConfig);
-                var serializer = new JavaScriptSerializer();
-                var version = (Dictionary<string, object>)serializer.DeserializeObject(result);
-                upgradePlatformVersion = version.ContainsKey("platform-version") ? (string)version["platform-version"] : null;
-            }
-            catch (Exception exception)
-            {
-                log.Warn($"Plugin call prepare_host_upgrade.getVersion on {Host.Name()} threw an exception.", exception);
+
+            var hotfix = HotfixFactory.Hotfix(Host);
+            if (hotfix != null && hotfix.ShouldBeAppliedTo(Host))
+                return new HostDoesNotHaveHotfixWarning(this, Host);

```

Comment:
This warning is now displayed for th eManual upgrade mode, but in this mode we do not require the RPU hotfixes to be installed prior to the upgrade.

(Same for the vSwitch controller check)